### PR TITLE
correct type of boolean operators

### PR DESCRIPTION
**Pull request by kroening**
- [ ] Each commit message has a non-empty body, explaining why the change was made.
- [X] My contribution is formatted in line with CODING_STANDARD.md.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [x] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [X] My PR is restricted to a single feature or bugfix.
- [X] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: regression/goto-analyzer/intervals_01/main.c [hunk 1]
```diff
@@ -1,3 +1,4 @@
+#include <assert.h>
```
Comment (tautschnig) [inline regression/goto-analyzer/intervals_01/main.c]: Ok, but how does this relate to the PR? Adding a body to the commit message might be really helpful to explain this.

File: src/ansi-c/c_typecheck_expr.cpp [hunk 2]
```diff
@@ -306,6 +306,8 @@ void c_typecheck_baset::typecheck_expr_main(exprt &expr)
     // replace declaration by symbol expression
     symbol_exprt bound=to_symbol_expr(expr.op0().op0());
     expr.op0().swap(bound);
+
+    implicit_typecast_bool(expr.op1());
```
Comment (tautschnig) [inline src/ansi-c/c_typecheck_expr.cpp]: Could we have a regression test for this? I suppose that might be fairly easy to construct from one of the existing tests using quantifiers?

File: regression/goto-analyzer/intervals_01/main.c (hunk 1 above)
Comment (kroening) [reply regression/goto-analyzer/intervals_01/main.c]: Unrelated -- will drop.

File: src/ansi-c/c_typecheck_expr.cpp (hunk 2 above)
Comment (kroening) [reply src/ansi-c/c_typecheck_expr.cpp]: Covered by existing tests; I've added a note about that into the commit message.

**Comment by smowton** [comment]
To check: will this now crash with `--no-simplify` because the backend doesn't know how to handle bool-to-int casts?

File: src/ansi-c/c_typecheck_expr.cpp (hunk 2 above)
Comment (tautschnig) [reply src/ansi-c/c_typecheck_expr.cpp]: Hmm, if the existing tests cover it, why does this change have no impact (such as turning a test from KNOWNBUG to CORE)?

**Comment by tautschnig** [comment]
@kroening I'm not sure whether MC/DC coverage is still handled correctly, these are now the only test failures left.

**Comment by TGWDB** [comment]
Closing due to age (no further comment on PR content), please reopen with rebase on develop if you intent to continue this work.

**Pull request closed without merging**